public/app/features/alerting/unified/components/rule-editor/FolderAndGroup.tsx

```diff
@@ -0,0 +1,236 @@
+
+  const groupOptions: Array<SelectableValue<string>> = mapGroupsToOptions(
+    useGetGroups(groupfoldersForGrafana?.result, folderTilte)
+  );
+  const groupsForFolder = groupfoldersForGrafana?.result;
+  return { groupOptions, groupsForFolder, loading: groupfoldersForGrafana?.loading };
+};
+
+const useRuleFolderFilter = (existingRuleForm: RuleForm | null) => {
+  const isSearchHitAvailable = useCallback(
+    (hit: DashboardSearchHit) => {
+      const rbacDisabledFallback = contextSrv.hasEditPermissionInFolders;
+
+      const canCreateRuleInFolder = contextSrv.hasAccessInMetadata(
+        AccessControlAction.AlertingRuleCreate,
+        hit,
+        rbacDisabledFallback
+      );
+
+      const canUpdateInCurrentFolder =
+        existingRuleForm &&
+        hit.folderId === existingRuleForm.id &&
+        contextSrv.hasAccessInMetadata(AccessControlAction.AlertingRuleUpdate, hit, rbacDisabledFallback);
+      return canCreateRuleInFolder || canUpdateInCurrentFolder;
+    },
+    [existingRuleForm]
+  );
+
+  return useCallback<FolderPickerFilter>(
+    (folderHits) =>
+      folderHits
+        .filter(isSearchHitAvailable)
+        .filter((value: DashboardSearchHit) => !containsSlashes(value.title ?? '')),
+    [isSearchHitAvailable]
+  );
+};
+function InfoIcon({ text }: { text: string }) {
+  return (
+    <Tooltip placement="top" content={<div>{text}</div>}>
+      <Icon name="info-circle" size="xs" />
+    </Tooltip>
+  );
+}
+
+export function FolderAndGroup({ initialFolder }: FolderAndGroupProps) {
+  const {
+    formState: { errors },
+    watch,
+    control,
+  } = useFormContext<RuleFormValues>();
+
+  const styles = useStyles2(getStyles);
+  const folderFilter = useRuleFolderFilter(initialFolder);
+  const dispatch = useDispatch();
+  const [isAddingGroup, setIsAddingGroup] = useState(false);
+
+  const folder = watch('folder');
+  const group = watch('group');
+  const [selectedGroup, setSelectedGroup] = useState(group);
+  const initialRender = useRef(true);
+
+  const { groupOptions, groupsForFolder, loading } = useGetGroupOptionsFromFolder(folder?.title ?? '');
+
+  useEffect(() => {
+    dispatch(fetchRulerRulesIfNotFetchedYet(GRAFANA_RULES_SOURCE_NAME));
+  }, [dispatch]);
+
+  const resetGroup = useCallback(() => {
+    if (group && !initialRender.current && folder?.title) {
+      setSelectedGroup('');
+    }
+    initialRender.current = false;
+  }, [initialRender, folder, setSelectedGroup, group]);
+
+  const groupIsInGroupOptions = useCallback(
+    (group_: string) => {
+      return groupOptions.includes((groupInList: SelectableValue<string>) => groupInList.label === group_);
+    },
+    [groupOptions]
+  );
+  const folderTilte = folder?.title ?? '';
+
+  useEffect(() => {
+    if (folderTilte && groupOptions && !groupIsInGroupOptions(selectedGroup)) {
+      setIsAddingGroup(false);
+      resetGroup();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
can you add all the dependencies to the array instead of this line disabling it?

Reply:
good idea! @VikaCep